### PR TITLE
add repository update frequency documentation

### LINKED ISSUE #6203: Document repo update behavior
We have had inquiries from customers and members of our own team about how frequently Sourcegraph updates repositories and if there is a way to control this behavior. Examples:

From [$CUSTOMER](https://app.hubspot.com/contacts/2762526/company/2150909975/):

> Need a quick reposes on this if you have solution.
> 
> We are using “Other Git repository hosts” to let SG clone the repos and then it check for updates.
>
> How frequent it tries to pull the repo updates ?
> 
> Can we reduce the frequency to reduce load on repo host ?
> And also if we need to temporarily shut-off the index update ?

From @ryan-blunden:

>  What is the expected delay from when Sourcegraph syncs a new commit, to that change being indexed and findable? ([Thread](https://sourcegraph.slack.com/archives/CHPC7UX16/p1571953374113500))

We have documentation for specific fields like `repoListUpdateInterval` and `gitMaxConcurrentClones`, but AFAICT no high-level documentation.

Proposal: Add to our docs (1) FAQs about repository update behavior and (2) a description of the mental model users should have when reasoning about repository update behavior.

### PR DESCRIPTION
**Pull request by slimsag**
Fixes #6203

### CODE REVIEW

File: doc/admin/repo/update_frequency.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Repository update frequency
+
+Sourcegraph polls code hosts to keep repository contents up to date by default, effectively periodically running `git pull`. You can also configure Sourcegraph to use [repository webhooks](webhooks.md) instead, but this is generally not needed.
```
Comment (beyang) [inline doc/admin/repo/update_frequency.md]:
```suggestion
By default, Sourcegraph polls code hosts to keep repository contents up to date, effectively running `git pull` periodically. You can also configure Sourcegraph to use [repository webhooks](webhooks.md), but this is usually not necessary.
```

File: doc/admin/repo/update_frequency.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+The frequency at which Sourcegraph polls the code host for updates is determined by a smart hueristic based on past commit frequency in the repository. For example, if a repositories last commit was 8 hours ago, then the next update will be scheduled 4 hours from now. If there are still no new commits, then the next update will be scheduled 6 hours from then.
```
Comment (uwedeportivo) [inline doc/admin/repo/update_frequency.md]: heuristic

Comment (beyang) [inline doc/admin/repo/update_frequency.md]:
```suggestion
The frequency at which Sourcegraph polls the code host for updates is determined by a smart heuristic based on past commit frequency in the repository. For example, if a repository's last commit was 8 hours ago, then the next sync will be scheduled 4 hours from now. If after 4 hours, there are still no new commits, then the next update will be scheduled 6 hours from then.
```

Comment (beyang) [reply doc/admin/repo/update_frequency.md]: Can we say what sort of backoff it is? Linear, exponential, or somewhere in between?

File: doc/admin/repo/update_frequency.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+Repositories will never be updated more frequently than 45 seconds, and no less frequently than every 8 hours.
+
+After Sourcegraph has polled the code host for Git updates, the search index will automatically update a short while after (usually taking just a few minutes per repository).
```
Comment (beyang) [inline doc/admin/repo/update_frequency.md]:
```suggestion
After Sourcegraph has updated a repository's Git data, the global search index will automatically update a short while after (usually a few minutes).
```

File: doc/admin/repo/update_frequency.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+## Limiting repository updates
+
+If you wish to control how frequently repositories are discovered or how frequently Sourcegraph polls your code host for updates, tuning parameters are available in the site configuration:
+
+- [repoListUpdateInterval](../config/site_config.md#repoListUpdateInterval) controls how frequently we check the code host _for new repositories_ in minutes. Defaults to 1 minute.
```
Comment (beyang) [inline doc/admin/repo/update_frequency.md]:
```suggestion
- [repoListUpdateInterval](../config/site_config.md#repoListUpdateInterval) controls how frequently we check the code host _for new repositories_ in minutes.
```

^ I think it's better to let the user click through to learn what the default is.

File: doc/admin/repo/update_frequency.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+- [gitMaxConcurrentClones](../config/site_config.md#gitMaxConcurrentClones) controls the maximum number of _concurrent_ cloning / pulling operations that Sourcegraph will perform.
+
+You may also choose to disable automatic Git updates entirely and instead [configure repository webhooks](webhooks.md).
```
Comment (nicksnyder) [inline doc/admin/repo/update_frequency.md]:
```suggestion
You may also choose to [disable automatic Git updates entirely](https://docs.sourcegraph.com/admin/config/site_config#disableAutoGitUpdates) and instead [configure repository webhooks](webhooks.md).
```

File: doc/admin/repo/update_frequency.md (hunk 2 above)
Comment (nicksnyder) [reply doc/admin/repo/update_frequency.md]: I would use the word `fetch` instead of `sync` or `update`

**Comment by slimsag** [comment]
Appreciate the help! I threw this together really quick so apologies for the typos you caught!

File: doc/admin/repo/update_frequency.md (hunk 5 above)
Comment (slimsag) [reply doc/admin/repo/update_frequency.md]: I intentionally didn't link to this, the webhook documentation already describes this in more depth.

File: doc/admin/repo/update_frequency.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# Repository update frequency
+
+By default, Sourcegraph polls code hosts to keep repository contents up to date, effectively running `git pull` periodically. You can also configure Sourcegraph to use [repository webhooks](webhooks.md), but this is usually not necessary.
+
+The frequency at which Sourcegraph polls the code host for updates is determined by a smart heuristic based on past commit frequency in the repository. For example, if a repository's last commit was 8 hours ago, then the next sync will be scheduled 4 hours from now. If after 4 hours, there are still no new commits, then the next update will be scheduled 6 hours from then.
```
Comment (slimsag) [inline doc/admin/repo/update_frequency.md]:
```suggestion
The frequency at which Sourcegraph polls the code host for updates is determined by a smart heuristic based on past commit frequency in the repository. For example, if a repository's last commit was 8 hours ago, then the next sync will be scheduled 4 hours from now. If after 4 hours, there are still no new commits, then the next sync will be scheduled 6 hours from then.
```

**Comment by slimsag** [comment]
@nicksnyder I converged on just `sync` -> `fetch` might imply we are always fetching the entire repository which isn't true (we're synching our copy of it); also we are not just doing this for Git data but also synchronizing our metadata about the repository

**Pull request merged**